docs/standard/microservices-architecture/multi-container-microservice-net-applications/implementing-api-gateways-with-ocelot.md

```diff
@@ -0,0 +1,566 @@
+
+}).AddJwtBearer(options =>
+{
+    options.Authority = identityUrl;
+    options.RequireHttpsMetadata = false;
+    options.Audience = "basket";
+});
+```
+
+ Now, if you try to access any secured microservice like the Basket microservice with a ReRoute URL based on the API Gateway like http://localhost:5202/api/v1/b/basket/1 then you’ll get a 401 Unauthorized unless you provide a valid token. On the other hand, if a ReRoute URL is authenticated, Ocelot will invoke whatever downstream scheme is associated with it (the internal microservice URL).
+
+**Authorization at Ocelot’s ReRoutes tier.**  Ocelot supports claims-based authorization evaluated after the authentication. You set the authorization at a route level by adding the following to you ReRoute configuration. 
+
+```csharp
+"RouteClaimsRequirement": {
+    "UserType": "employee"
+}
+```
+
+In that example, when the authorization middleware is called, Ocelot will find if the user has the claim type “UserType” in the token and if the value of that claim is “employee”. If it isn’t then the user will not be authorized and the response will be 403 forbidden. 
+
+## Using Kubernetes Ingress plus Ocelot API Gateways
+
+When using Kubernetes (like in an Azure Kubernetes Service cluster) you usually unify all the http requests through the [Kuberentes Ingress tier](https://kubernetes.io/docs/concepts/services-networking/ingress/) based on *Nginx*. 
+
+In Kuberentes, if you don’t use any ingress approach, then your services and pods have IPs only routable by the cluster network. 
+
+But if you use an ingress approach, you will have a middle tier in between the Internet and your services (including your API Gateways), acting as a reverse proxy.
+
+As a definition, an Ingress is a collection of rules that allow inbound connections to reach the cluster services. An ingress is usually configured to provide services externally-reachable URLs, load balance traffic, SSL termination and more. Users request ingress by POSTing the Ingress resource to the API server.
+
+In eShopOnContainers, when developing locally and using just your development machine as the Docker host, you are not using any ingress but only the multiple API Gateways. But when moving to a “production” environment based on Kuberentes, eShopOnCOntainers is using an ingress in front of the API gateways so that the clients still call the same base URL but the requests are routed to multiple API Gateways or BFF. 
+
+Note that the API Gateways are front-ends of facades only for the services or Web APIs but not for the web applications plus they might hide certain internal microservices. The ingress, however, is just redirecting http requests but not trying to hide anything.
+
+Therefore, having an ingress Nginx tier in Kuberentes in front of the web applications plus the several Ocelot API Gateways / BFF is the ideal architecture, as shown in the following diagram.
+
+ ![](./media/image41.png)
```

Comment:
images should be put in a directory with the article name inside media
and image names should also use friendly names to help with SEO too